runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/windmill/GrpcWindmillServerTest.java

```diff
@@ -613,11 +694,14 @@ public void onCompleted() {
       }
     }
     stream.flush();
+    stream.close();
+
+    Thread.sleep(100);
```

Comment:
Instead of just sleeping could you poll some counter you increment when returning server errors, and wait for a couple errors to be returned?  That would help ensure the error handling path on the client actually runs.